pkg/controller/scyllacluster/resource.go

```diff
@@ -526,37 +526,52 @@ func StatefulSetForRack(r scyllav1.RackSpec, c *scyllav1.ScyllaCluster, existing
 							// TODO: unprivileged entrypoint
 							Command: func() []string {
 								cmd := []string{
-									path.Join(naming.SharedDirName, "scylla-operator"),
-									"sidecar",
-									fmt.Sprintf("--feature-gates=%s", func() string {
+									"/usr/bin/bash",
+									"-euEo",
+									"pipefail",
+									"-O",
+									"inherit_errexit",
+									"-c",
+									strings.TrimSpace(`
+printf 'INFO %s ignition - Waiting for /mnt/shared/ignition.done\n' "$( date '+%Y-%m-%d %H:%M:%S,%3N' )" > /dev/stderr
```

Comment:
What if scylla is ignited, starts and then crashes? This file won't be removed so container starts right away, but Scylla container changed it's ID and might not be tuned anymore. Am I right or I'm missing something?

Reply:
Do we want to wait for tuning in that case or rather bring ScyllaDB back asap?

Reply:
Restarted container may get different CPU and tuning needs to update CPU mask for IRQ pinning. Since Scylla startup script runs the benchmark at the startup I guess we want to wait for tuning to complete before we evaluate resources.

Reply:
We want to wait not only for tuning, but for values we wait in ignition (IPs), as these are still used in sidecar to generate scylla parameters without checking if they are available.

Reply:
>  tuning needs to update CPU mask for IRQ pinning

I does. My point was that ScyllaDB coming **back** on 90% of performance immediately and after 2 minutes getting to a 100% is better then waiting 2 minutes to start ScyllaDB at 100% of performance. IOW a degraded node is better then none and IRQ pinning is done asynchronously. Also the independent ignition probe will still show the pod as not ready so it will block the rollout or evictions until re-tuned.

> Since Scylla startup script runs the benchmark at the startup

The volume / disk is still the same, already tuned and the results are cached.
https://github.com/scylladb/scylla-operator/blob/efaff1235eb0ac468324bb3f1ca2f376146bd61a/pkg/sidecar/config/config.go#L302

What benchmark do you have in mind?

> for values we wait in ignition (IPs), as these are still used in sidecar to generate scylla parameters without checking if they are available.

It has run before so they were there already. Checking the service LB IP is and *has been* always racy, if someone messes up with the services in the meantime. 

Is there something you think the Scylla entrypoint wouldn't get through or something that wouldn't be re-tuned while ScyllaDB can already be restarting/serving?

Reply:
(guess I can remove the file in a preStop hook or check the dates but restarting faster on failures seemed like a good thing)

Reply:
It all depends when crash happens. If it's before the benchmark, then we want to wait until tuning is completed, as we don't want to have skewed results stored in the cache and then used as source of truth by Scylla. 
If it's past the benchmark, then it make sense to start Scylla without waiting for tuning to be completed. 

But we also wait for Service and Pod IPs - for a reason - Pod IP is ephemeral and is not available in Status immediately after Pod starts.  Not having it, would result in empty parameter which would cause another crash. 

IMO we shouldn't differentiate whether benchmark was completed or not. The right way is to wait for all preconditions to be completed before starting Scylla. Removing ignition file in the preStop makes sense.

Reply:
> But we also wait for Service and Pod IPs - for a reason - Pod IP is ephemeral and is not available in Status immediately after Pod starts.

That's not the case of a container restart though. The IP is still the same. If a pod is deleted the ignition file goes away with the ephemeral storage.

> If it's past the benchmark, then it make sense to start Scylla without waiting for tuning to be completed.

You are correct that this would have to check the presence of io_tune results (if tuning is enabled) as it can fail during the generation. Let's leave it out for now and I'll update the preStop hook.

Reply:
added the rm